plugins/global-search/src/utils/indexer/IndexerProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useMemo, useState } from "react"
+import { IndexerContext } from "./context"
+import type { Events } from "./event-emitter"
+import { GlobalSearchIndexer } from "./indexer"
+import type { IndexEntry } from "./types"
+
+/**
+ * Creates an indexer instance and provides it to the children.
+ * Manages the index in a React state, so that it can use reactivity to show the results
+ */
+export function IndexerProvider({ children }: { children: React.ReactNode }) {
+    const indexer = useMemo(() => new GlobalSearchIndexer(), [])
+    const [isIndexing, setIsIndexing] = useState(false)
+    const [index, setIndex] = useState<Record<string, IndexEntry>>({})
+
+    useEffect(() => {
+        indexer.start()
+
+        const onUpsert = ({ entry }: Events["upsert"]) => {
+            setIndex(prev => ({ ...prev, [entry.id]: entry }))
```

Comment:
This seems like a non-urgent update that may be expensive, I wonder if it would be nice to wrap that in `startTransition`

Reply:
Also just curious if there's any reason to prefer syncing the entries from the Indexer with React in a `useEffect` over `useSyncExternalStore` (atp I'm not sure anymore what is the recommended approach, just curious)

Reply:
`startTransition`: Yes! That's a great idea, didn't thought about using it yet, as I don't show the data 😅 

`useSyncExternalStore `: Thought about it, that's why I added the event emitter already. But I'm not entirely sure how the filterer will work and what API it will have, and how the two might be connected.